### PR TITLE
Postgres source: add integration with data dog

### PR DESCRIPTION
**Pull request by sashaNeshcheret**
## What
Add datadog support for postgres source 

## How
We added dd-agent to postgres source. All other work will be provided from platform side
Added utils for tracing, and marked needed methods with @Trace

## Recommended reading order
1. `airbyte-integrations/bases/base-java/Dockerfile`
2. `ApmTraceUtils.java`
3. other files

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java
Comment (jdpgrailsdev) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java:145]: This should probably be a static constant to avoid creating it each time.

File: airbyte-integrations/connectors/source-postgres/Dockerfile
Comment (jdpgrailsdev) [inline airbyte-integrations/connectors/source-postgres/Dockerfile:21]: This should probably be set to the real version number/revert the change to the label below before merging.  We probably also need to update whatever script updates the release number to contain the `DD_VERSION` as a line that also needs to be modified when releasing a new version.

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java
Comment (jdpgrailsdev) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:86]: This is the name of the operations in the workers, which are Temporal activities.  I suspect that we will want a different operation name for the traces in here.  I'm not sure what we call the operations performed by a source, but whatever it is, that is what we should use for the operation name instead of activity.

File: airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java
Comment (jdpgrailsdev) [inline airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java:114]: We also need a filter somewhere in `KubePodProcess` to ensure that this env var is only passed to the source connector image IF it's one that has the agent present.

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java
Comment (jdpgrailsdev) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:97]: Nit:  the other operations use lowercase dash format.  Consider changing this one to be consistent with the others (or change them to be consistent with this lower camel case).

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:  LGTM with just a minor nit about operation name case consistency

**Comment by subodh1810** [comment]
@sashaNeshcheret we are not going to merge this PR right (not going to release the image to cloud). I assume you published the image just for your own testing?

**Review by subodh1810 — CHANGES_REQUESTED** [review]
See my comment https://github.com/airbytehq/airbyte/pull/21533#issuecomment-1460471994

**Comment by subodh1810** [comment]
@sashaNeshcheret cool thanks for clarifying :D

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java
Comment (akashkulk) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:287]: Same : do we need this annotation for every method?

File: airbyte-integrations/connectors/source-postgres-strict-encrypt/Dockerfile
Comment (akashkulk) [inline airbyte-integrations/connectors/source-postgres-strict-encrypt/Dockerfile:19]: Why is this bumped to `2.0.8-dd`. Shouldn't this be `2.0.4`?

**Comment by akashkulk** [comment]
Can you add the loom demoing this in the PR description? Thanks!

**Review by akashkulk — APPROVED** [review]

**Review by subodh1810 — APPROVED** [review]
approving cause PR cant be merged without my approval

File: airbyte-integrations/bases/base-java/Dockerfile
Comment (evantahler) [inline airbyte-integrations/bases/base-java/Dockerfile:10]: Does this mean that we are using the in-process collector or will we be forwarding metrics to the DD collector pod?

**Review by evantahler — COMMENTED** [review]
Nice work!

I'd like to chat a little about how tracing is enabled - do we pass the connector an API Key or a Collector URL?  Is there a related Platform PR that handles this?

I'm also curious as to why we went with DD directly, and not open-telemetry to keep the connectors themself as agnostic as possible.

**Pull request merged**
